drivers/SmartThings/matter-thermostat/src/init.lua

```diff
@@ -1520,150 +1525,81 @@ local function min_deadband_limit_handler(driver, device, ib, response)
+                                                        capabilities.airConditionerFanMode.fanMode("low"),
+                                                        capabilities.airPurifierFanMode.airPurifierFanMode.low(),
+                                                        capabilities.thermostatFanMode.thermostatFanMode.on() },
+    [clusters.FanControl.attributes.FanMode.MEDIUM] = { capabilities.fanMode.fanMode.medium(),
+                                                        capabilities.airConditionerFanMode.fanMode("medium"),
+                                                        capabilities.airPurifierFanMode.airPurifierFanMode.medium(),
+                                                        capabilities.thermostatFanMode.thermostatFanMode.on() },
+    [clusters.FanControl.attributes.FanMode.HIGH]   = { capabilities.fanMode.fanMode.high(),
+                                                        capabilities.airConditionerFanMode.fanMode("high"),
+                                                        capabilities.airPurifierFanMode.airPurifierFanMode.high(),
+                                                        capabilities.thermostatFanMode.thermostatFanMode.on() },
+    [clusters.FanControl.attributes.FanMode.ON]     = { capabilities.fanMode.fanMode.auto(),
+                                                        capabilities.airConditionerFanMode.fanMode("auto"),
+                                                        capabilities.airPurifierFanMode.airPurifierFanMode.auto(),
+                                                        capabilities.thermostatFanMode.thermostatFanMode.on() },
+    [clusters.FanControl.attributes.FanMode.AUTO]   = { capabilities.fanMode.fanMode.auto(),
+                                                        capabilities.airConditionerFanMode.fanMode("auto"),
+                                                        capabilities.airPurifierFanMode.airPurifierFanMode.auto(),
+                                                        capabilities.thermostatFanMode.thermostatFanMode.auto() },
+    [clusters.FanControl.attributes.FanMode.SMART]  = { capabilities.fanMode.fanMode.auto(),
+                                                        capabilities.airConditionerFanMode.fanMode("auto"),
+                                                        capabilities.airPurifierFanMode.airPurifierFanMode.auto(),
+                                                        capabilities.thermostatFanMode.thermostatFanMode.auto() }
+  }
+  local fan_mode_idx = device:supports_capability_by_id(capabilities.fanMode.ID) and 1 or
+    device:supports_capability_by_id(capabilities.airConditionerFanMode.ID) and 2 or
+    device:supports_capability_by_id(capabilities.airPurifierFanMode.ID) and 3 or
+    device:supports_capability_by_id(capabilities.thermostatFanMode.ID) and 4
+  if fan_mode_idx and fan_mode_idx >= 1 and fan_mode_idx <= 4 and fan_mode_event[ib.data.value][fan_mode_idx] then
```

Comment:
```suggestion
  if fan_mode_idx ~= false then
```
I think the rest of this is superfluous based on the handling- since the rest of these will _always_ be true if this ^ is true.